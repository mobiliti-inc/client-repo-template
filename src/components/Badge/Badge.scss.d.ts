// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'available': string;
  'coming-soon': string;
  'fades-in-out': string;
  'unavailable': string;
}
declare var cssExports: CssExports;
export = cssExports;
