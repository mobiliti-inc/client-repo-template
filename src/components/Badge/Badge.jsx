import React from 'react';

import styles from './Badge.scss';
import StatusMessages from '../../constants/vehicleStatusCodes';

class Badge extends React.PureComponent {
	static propTypes = {
		status: PropTypes.number.isRequired
	}

	renderStatusMessage = status => {
		const message = StatusMessages[status];
		return message;
	};

	render() {
		const { status } = this.props;
		if (typeof status !== 'number') return null;
		const styleName = status === 1 ? 'available' : 'unavailable';
		const comingSoon = status === 3 ? 'coming-soon' : '';
		return (
			<div styleName={`${styleName} ${comingSoon}`}>{this.renderStatusMessage(status)}</div>
		);
	}
}

export default CSSModules(Badge, styles, { allowMultiple: true });
