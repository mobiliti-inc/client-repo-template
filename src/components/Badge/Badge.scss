@import "../../styles/_variables.scss";

.available {
	color: #fff;
	background: #26A86F;
	border-radius: 4px;
	font-family: $hanken-sans;
	font-size: 14px;
	padding: 6px 0;
	text-align: center;
	width: 80px;
	height: 30px;
	text-transform: capitalize;
	font-weight: 500;
	margin: 5px 0;
}

.unavailable {
	composes: available;
	background: #8f8f8f;
	width: 150px;
}

.coming-soon {
	width: 110px;
}