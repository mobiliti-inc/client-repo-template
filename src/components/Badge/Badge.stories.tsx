import { storiesOf } from '@storybook/react';
import { number } from '@storybook/addon-knobs';
import * as React from 'react';
import { Badge } from '../';

storiesOf('Badge', module)
	.add('default', () =>
		<Badge status={number('status', 1)} />, {
			notes: 'status can be 1, 2 or 3'
		}
	);
