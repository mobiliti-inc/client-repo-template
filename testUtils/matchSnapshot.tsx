const { shallowToJson } = require('enzyme-to-json');

export const matchSnapshot = (subject: any) => {
	expect(shallowToJson(subject)).toMatchSnapshot();
};
