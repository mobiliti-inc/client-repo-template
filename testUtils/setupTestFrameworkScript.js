const React = require("react");
// const CSSModules = require("react-css-modules");
const Enzyme = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

const { shallow, render, mount } = Enzyme;

Enzyme.configure({ adapter: new Adapter() });
jest.setTimeout(30000);

global.React = React;
// global.CSSModules = CSSModules;
global.shallow = shallow;
global.render = render;
global.mount = mount;
