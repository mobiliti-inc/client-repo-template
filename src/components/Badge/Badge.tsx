import * as React from 'react';

import { StatusMessages } from '../../utils';
import './Badge.scss';

interface BadgeProps {
	status: number;
}

const Badge: React.FC<BadgeProps> = (props) => {
	const { status } = props;

	const renderStatusMessage = (status: any) => {
		const message = StatusMessages[status];
		return message;
	};

	if (typeof status !== 'number') return null;
	const styleName = status === 1 ? 'available' : 'unavailable';
	const comingSoon = status === 3 ? 'coming-soon' : '';
	return (
		<div styleName={`${styleName} ${comingSoon}`}>{renderStatusMessage(status)}</div>
	);
};

export default Badge;
