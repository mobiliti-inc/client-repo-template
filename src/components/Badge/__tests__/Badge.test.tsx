import React from 'react';

import Badge from '../Badge';
import { shallow } from 'enzyme';

import { matchSnapshot } from '../../../../testUtils/matchSnapshot';

describe('Badge', () => {
	it('renders', () => {
		const subject = shallow(<Badge />);
		matchSnapshot(subject);
	});
});
