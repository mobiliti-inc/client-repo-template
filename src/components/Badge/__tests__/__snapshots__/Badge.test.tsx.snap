// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders 1`] = `""`;
